l10n_it_withholding_tax/models/account.py

```diff
@@ -62,7 +63,12 @@ def create(self, vals):
             # Note that this is always executed, for every reconciliation.
             # Thus, we must not change amount when not in withholding tax case
             amount = vals.get('amount_currency') or vals.get('amount')
-            if amount > invoice.amount_net_pay:
+            digits_rounding_precision = invoice.company_id.currency_id.rounding
+            if float_compare(
+                amount,
+                invoice.amount_net_pay,
+                precision_rounding=digits_rounding_precision
+            ) == 1:
```

Comment:
E `amount < invoice.amount_net_pay` non può capitare?

Reply:
Non saprei, mi sono limitato ad adattare il codice esistente.

Reply:
Il codice esistente esegue `vals.update` solo quando `amount > invoice.amount_net_pay`, mentre in questa PR viene eseguito anche se `amount < invoice.amount_net_pay`.
O leggo male?

Reply:
float_compare() non è una funzione di comparazione come l'argomento  di https://docs.python.org/3/library/functools.html#functools.cmp_to_key ?

Ossia se amount > invoice.amount_net_pay  allora float_compare(amount, invoice.amount_net_pay, ...) restituisce 1 (tecnicamente >0).  Se amount < invoice.amount_net_pay restituisce -1 (<0). Se sono uguali restituisce 0.

Reply:
@eLBati https://github.com/odoo/odoo/blob/8f6430359b347d3669a536eca0a297c66ce99163/odoo/tools/float_utils.py#L153-L154

Reply:
Ah OK, grazie